packages/next/src/server/base-server.ts

```diff
@@ -1767,16 +1767,16 @@ export default abstract class Server<
       return builtinRequestContext.waitUntil
     }
 
-    if (process.env.__NEXT_TEST_MODE) {
-      // we're in a test, use a no-op.
-      return Server.noopWaitUntil
-    }
```

Comment:
this is an unfortunate leftover of early experiments, and it caused some tests for `after` to pass even though they shouldn't

in particular, `after` was actually throwing in `edge` pages in next dev/next start (because `waitUntil` wasn't actually passed into the edge sandbox), but this bit hid that error in tests.